docs/development/quickstart.rst

```diff
@@ -0,0 +1,365 @@
+`GitHub documentation for contributing to projects
+<https://docs.github.com/en/get-started/quickstart/contributing-to-projects>`_. GitHub
+provides a quick tutorial using a test repository that may help you become more familiar
+with forking a repository, cloning a fork, creating a feature branch, pushing changes
+and making pull requests.
+
+Creating a feature branch
+-------------------------
+
+Your local ``main`` branch should always reflect the current state of astropy repository.
+First ensure it's up-to-date with the main astropy repository::
+
+    git checkout main
+    git pull upstream main --ff-only
+
+Now create a feature branch for making your changes. For example::
+
+    git switch -c subpackage-new-feature
+
+This changes your working branch from ``main`` to the ``subpackage-new-feature`` branch.
+Keep any changes in this branch specific to one bug or feature so it is clear what the
+branch brings to astropy. You can have many feature branches and switch in between them
+using the ``git switch`` command.
+
+Using a descriptive branch name can help you stay organized. For example
+```io-ascii-commented-header-15513``` might be a good name for a branch that fixes the
+commented header issue `#15513 <https://github.com/astropy/astropy/issues/15513>`_ in
+the ``io.ascii`` sub-package.
+
+.. Important:: Never use the ``main`` branch for making changes. Always create a new
+   feature branch for your changes.
+
+When you want to update the feature branch with changes in main after
+you created the branch, check the section on
+:ref:`updating a PR <contributing.update-pr>`.
+
+.. _contributing.commit-code:
+
+Making code or documentation changes
+------------------------------------
+
+Now comes the fun part where you use your favorite editor or IDE to make changes to the
+code or documentation! At a high level this breaks into a few parts:
+
+- **Make changes**: Make the changes you want to make. This could be fixing a bug,
+  adding a new feature, or updating the documentation.
+- **Test changes**: For code changes, ensure that they work as expected following the
+  process outlined in the :ref:`testing-guidelines` section.
+- **Build documentation**: If you are updating the documentation, you will want to
+  :ref:`build the documentation <builddocs>` to ensure that it looks good.
+- **Add a changelog entry**: For most code changes you will need to
+  :ref:`add-changelog`.
+
+.. tip:: For more information and examples see :ref:`edit-flow` section.
+
+You can see all the changes you've currently made by running:
+
+.. code-block:: shell
+
+    git status
+
+You can then commit your all your changes to your local repository with an explanatory
+commit message:
+
+.. code-block:: shell
+
+    git add files-that-you-changed ...
+    git commit -m "your commit message goes here"
+
+You should make frequent commits and always include a commit message. Each commit
+should represent one logical set of changes.
```

Comment:
Good point. This is a bit subtle and complex, and feels out-of-scope at this level. I just removed it since our quickstart guide is not going to teach people how to be effective developers. Instead just get them through a PR and they will start to learn with feedback from maintainers.